iib/workers/tasks/build.py

```diff
@@ -426,11 +427,21 @@ def _get_present_bundles(from_index, base_dir):
 
     # If no data is returned there are not bundles present
     if not bundles:
-        return []
+        return [], []
 
     # Transform returned data to parsable json
+    unique_present_bundles = []
+    unique_present_bundles_pull_spec = []
     present_bundles = [json.loads(bundle) for bundle in re.split(r'(?<=})\n(?={)', bundles)]
-    return present_bundles
+
+    for bundle in present_bundles:
+        bundle_path = bundle['bundlePath']
+        if bundle_path in unique_present_bundles_pull_spec:
+            continue
+        unique_present_bundles.append(bundle)
+        unique_present_bundles_pull_spec.append(bundle_path)
+
+    return unique_present_bundles, unique_present_bundles_pull_spec
```

Comment:
Hmm, I personally find it confusing that this function returns two structures where one is the subset of the other. My preference would be to keep this part as it was. I wonder what others think.

Reply:
The only reason to do this was to avoid re-doing the same computation later when we want just the pull_specs of the labels. I figured we can create the list when we are filtering unique bundles and then use it in the rest of the code rather than doing the same computation over and over.

Reply:
So can I please suggest that we keep it this way

Reply:
Yeah, I understand the reasoning, but I feel like we are sacrificing design and readability to gain minuscule amount of performance. But I don't have strong feelings about it, I'm okay with keeping it as-is.

Reply:
I personally think it's not too bad of a trade off since it's just this function. I'll keep it as-is for now. :)